kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -156,22 +157,26 @@ public static void verifyDeltaVersionsContiguous(List<Long> versions, Path table
 
   private SnapshotImpl createSnapshot(
       LogSegment initSegment, Engine engine, SnapshotQueryContext snapshotContext) {
-    // Note: LogReplay now loads the protocol and metadata (P & M) only when invoked (as opposed to
-    //       eagerly in its constructor). Nonetheless, we invoke it right away, so SnapshotImpl is
-    //       still constructed with an "eagerly"-loaded P & M.
+    final ProtocolMetadataLogReplay.Result protocolMetadataResult =
+        ProtocolMetadataLogReplay.loadProtocolAndMetadata(
+            engine, initSegment, tablePath, snapshotContext.getSnapshotMetrics());
 
     final LogReplay logReplay =
```

Comment:
Actually ... we should just NOT pass in LogReplay ... we should just pass in the crcInfo

And this crcInfo can be re-used when we create our PostCommitSnapshot!